conjure-go-client/httpclient/internal/request_retrier.go

```diff
@@ -45,11 +45,13 @@ type RequestRetrier struct {
 // NewRequestRetrier creates a new request retrier.
 // Regardless of maxAttempts, mesh URIs will never be retried.
 func NewRequestRetrier(uris []string, retrier retry.Retrier, maxAttempts int) *RequestRetrier {
-	offset := rand.Intn(len(uris))
+	offset := 0
+	urisCopy := append([]string(nil), uris...)
```

Comment:
Lets avoid allocating the empty slice
```suggestion
	urisCopy := make([]string, 0, len(uris))
	copy(urisCopy, uris)
```